apps/state/lib/state/connecting_stops.ex

```diff
@@ -0,0 +1,192 @@
+    |> MapSet.difference(MapSet.new(override_stops(overrides, stop, :remove)))
+    |> MapSet.to_list()
+  end
+
+  # Find "standalone" stops close to a given stop's entrances, if it has entrances, otherwise
+  # close to the stop itself.
+  @spec standalone_stops_around(Stop.t()) :: [Stop.t()]
+  defp standalone_stops_around(%{id: stop_id, latitude: latitude, longitude: longitude}) do
+    case entrances(stop_id) do
+      [] ->
+        State.Stop.around(latitude, longitude, @radius) |> Enum.filter(&Stop.standalone?/1)
+
+      entrances ->
+        entrances |> Stream.flat_map(&standalone_stops_around/1) |> Enum.uniq()
+    end
+  end
+
+  @spec entrances(Stop.id()) :: [Stop.t()]
+  defp entrances(parent_station) do
+    State.Stop.Cache.match(%{location_type: 2, parent_station: parent_station}, :parent_station)
+  end
+
+  # Get stops that should be added/removed from a given parent station's connections.
+  @spec override_stops(overrides, Stop.t(), :add | :remove) :: [Stop.t()]
+  defp override_stops(overrides, %{id: stop_id}, kind) when kind in [:add, :remove] do
+    overrides |> Map.get(stop_id, %{}) |> Map.get(kind, []) |> State.Stop.by_ids()
+  end
+
+  # Return all rotations of a non-empty list.
+  # Example: [1, 2, 3] => [[1, 2, 3], [2, 3, 1], [3, 1, 2]]
+  @spec rotations([...]) :: [[...]]
+  defp rotations([item]), do: [[item]]
+
+  defp rotations([_ | _] = items) do
+    0..(length(items) - 1)
+    |> Enum.map(fn count ->
+      {before, [first | rest]} = Enum.split(items, count)
+      [first | before ++ rest]
+    end)
+  end
+
+  # Given an enumerable of lists, split them into heads and tails, group tails that share a head,
+  # then flatten and uniq the tails for each head.
+  # Example: [[1, 2, 3], [1, 3, 4], [2, 3, 4]] => [{1, [2, 3, 4]}, {2, [3, 4]}]
+  @spec group_and_merge_by_head(Enumerable.t()) :: Enumerable.t()
+  defp group_and_merge_by_head(combinations) do
+    combinations
+    |> Enum.group_by(&hd/1, &tl/1)
+    |> Stream.map(fn {head, tail} -> {head, tail |> List.flatten() |> Enum.uniq()} end)
+  end
+
+  # Given a stop and a list of possible connecting stops, sorts the connections by distance from
+  # the "base" stop, then rejects any that only serve route patterns also served by the base stop
+  # or a closer connection.
+  @spec reject_redundant_connections(Stop.t(), [Stop.t()]) :: [Stop.t()]
+  defp reject_redundant_connections(from_stop, connecting_stops) do
+    connecting_stops
+    |> Enum.sort_by(GeoDistance.cmp(from_stop.latitude, from_stop.longitude))
+    |> Stream.map(fn stop -> {stop, patterns_at_stop(stop)} end)
+    |> Enum.reduce({[], patterns_at_stop(from_stop)}, fn
+      {stop, stop_patterns}, {stops, seen_patterns} ->
+        if MapSet.subset?(stop_patterns, seen_patterns),
+          do: {stops, seen_patterns},
+          else: {[stop | stops], MapSet.union(seen_patterns, stop_patterns)}
+    end)
+    |> elem(0)
+    |> Enum.reverse()
```

Comment:
Can you avoid the `reverse` by sorting the stops in reverse order on 176?

Reply:
I think we need to examine the stops in ascending distance order so the closest stop (of multiple that share the same route patterns) is retained. If we reverse-sorted on 176, then the furthest-away stop would be kept instead.